Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java

```diff
@@ -0,0 +1,389 @@
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import timber.log.Timber;
+
+/**
+ * Created by chavi on 10/19/15.
+ */
+public class GeoTagImagesService extends Service {
+    private static final String SSH_USERNAME = "root";
+    private static final String SSH_PASSWORD = "TjSDBkAu";
+    private static final SshConnection soloSshLink = new SshConnection(BuildConfig.SOLO_LINK_IP, SSH_USERNAME, SSH_PASSWORD);
+    private static final String CAMERA_TLOG_FILE = "/log/camera_msgs.tlog";
+    private static final int NUMBER_OF_RETRIES = 3;
+    private static final String GEO_TAG_ROOT_NAME = "GeoTag";
+
+    private static final String PACKAGE_NAME = Utils.PACKAGE_NAME;
+
+    public static final String ACTION_START_LOADING_LOGS = PACKAGE_NAME + ".action.START_LOADING_LOGS";
+    public static final String ACTION_CANCEL_LOADING_LOGS = PACKAGE_NAME + ".action.CANCEL_LOADING_LOGS";
+    public static final String ACTION_START_GEOTAGGING = PACKAGE_NAME + ".action.START_GEOTAGGING";
+    public static final String ACTION_CANCEL_GEOTAGGING = PACKAGE_NAME + ".action.CANCEL_GEOTAGGING";
+
+    public static final String STATE_FINISHED_LOADING_LOGS = PACKAGE_NAME + ".FINISHED_LOADING_LOGS";
+    public static final String STATE_FINISHED_GEOTAGGING = PACKAGE_NAME + ".FINISHED_GEOTAGGING";
+    public static final String STATE_PROGRESS_UPDATE_GEOTAGGING = PACKAGE_NAME + ".PROGRESS_UPDATE_GEOTAGGING";
+
+    public static final String EXTRA_SUCCESS = PACKAGE_NAME + ".EXTRA_SUCCESS";
+    public static final String EXTRA_FAILURE_MESSAGE = PACKAGE_NAME + ".EXTRA_FAILURE_MESSAGE";
+    public static final String EXTRA_PROGRESS = PACKAGE_NAME + ".EXTRA_PROGRESS";
+    public static final String EXTRA_TOTAL = PACKAGE_NAME + ".EXTRA_TOTAL";
+    public static final String EXTRA_GEOTAGGED_FILES = PACKAGE_NAME + ".EXTRA_GEOTAGGED_FILES";
+
+    private AsyncTask getTlogsTask;
+    private GeoTagTask geoTagTask;
+    private LocalBroadcastManager lbm;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        lbm = LocalBroadcastManager.getInstance(getApplicationContext());
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        if (intent == null) {
+            return super.onStartCommand(intent, flags, startId);
+        }
+
+        switch (intent.getAction()) {
+            case ACTION_START_LOADING_LOGS:
+                startLoadingLogs();
+                break;
+            case ACTION_CANCEL_LOADING_LOGS:
+                cancelLoadingLogs();
+                break;
+            case ACTION_START_GEOTAGGING:
+                startGeoTagging();
+                break;
+            case ACTION_CANCEL_GEOTAGGING:
+                cancelGeoTagging();
+                break;
+        }
+        return super.onStartCommand(intent, flags, startId);
```

Comment:
@ne0fhyk True, but the super does handle START_STICKY and START_STICKY_COMPATIBILITY which is what I want. I can return START_STICKY explicitly if you want